### PR TITLE
enhance: incrementally update delegator snapshots

### PR DESCRIPTION
**Pull request by sijie-ni-0214**
issue: #50334

### What changed

- Track pending delegator distribution changes as sealed/growing snapshot deltas.
- Apply sealed distribution updates incrementally in the async snapshot loop instead of rebuilding the full delegator snapshot for every change.
- Return an already-closed wait signal for no-op removal requests so stale release requests cannot block waiting for a snapshot update that will never run.
- Keep full synchronous snapshot rebuilds for paths that require immediate consistency, including growing segment updates, target version sync, removals, and initialization.

In an internal high-concurrency QN recovery benchmark, this optimization improved recovery speed by about 30%.

### Tests

- `gofumpt -l internal/querynodev2/delegator/distribution.go internal/querynodev2/delegator/distribution_test.go`
- `git diff --check upstream/master..HEAD`
- `go test -tags dynamic,test -gcflags="all=-N -l" -count=1 ./internal/querynodev2/delegator -run 'TestDistributionSuite/TestRemoveDistributionNoOpReturnsClosedSignal|TestDistributionSuite/TestRemoveDistribution'` blocked locally because `pkg-config` cannot find `milvus_core.pc`, `rocksdb.pc`, and `milvus-storage.pc`.

### CODE REVIEW

**Comment by czs007** [comment]
A couple of non-blocking suggestions on the incremental snapshot path (independent of the current `ut-go` failure / hold):

**(strongly recommended) Add a randomized differential test: incremental vs. full rebuild.** The whole design rests on the invariant that the incremental `applySnapshotDeltaLocked` path and the full `genSnapshot` path always produce an equivalent snapshot — and that invariant is what guards search correctness. The current unit tests only cover fixed scenarios; there's no coverage of randomly interleaved operations. Suggest a property/fuzz test that applies a random sequence of `AddDistributions` / `RemoveDistributions` / `MarkOfflineSegments` / `AddGrowing` / `SyncTargetVersion`, then asserts that the incrementally-maintained state (`snapshotSegments` / `snapshotNodeSegments` + the published `dist`) equals a full `genSnapshot` over the same `sealedSegments` / `growingSegments`. This kind of test catches position-index / `affectedNodes` corner cases that are hard to spot by reading the code, and it's the highest-value addition here.

**(nits)**
- `applySnapshotDeltaLocked` (`distribution.go:412-414`): the `last == nil` fallback calls `genSnapshot()` and returns `nil, nil`, so the subsequent `updateServiceableByChangedSegments(nil, nil)` returns early and serviceability is not refreshed — inconsistent with the "genSnapshot is always followed by updateServiceable" convention used elsewhere. It looks unreachable in practice (the constructor generates the first snapshot before any notification), so either add `updateServiceable` here too, or add a comment noting the branch is theoretically unreachable.
- `snapshotDelta.growingUpserts` (`distribution.go:154` + the branch in `applySnapshotDeltaLocked`): never written — `AddGrowing` goes through a synchronous full `genSnapshot` and doesn't record a growing upsert delta, so the `growingUpserts` branch is dead code. Either drop the field + branch, or add a comment that growing inserts intentionally take the synchronous full-rebuild path and the field is reserved.
- `getClosedCh` (`distribution.go:55`): the `sync.Once` lazy init of an always-closed shared channel is a bit over-engineered; a package-level `var closedCh = func() chan struct{} { ch := make(chan struct{}); close(ch); return ch }()` would be simpler. No correctness impact.

File: internal/querynodev2/delegator/distribution.go [hunk 1]
```diff
@@ -350,18 +609,20 @@ func (d *distribution) Serviceable() bool {
 func (d *distribution) updateServiceable(triggerAction string) {
 	loadedSealedSegments := int64(0)
 	totalSealedRowCount := int64(0)
-	unloadedSealedSegments := make([]SegmentEntry, 0)
+	unloadedSealedSegments := make(map[UniqueID]SegmentEntry)
 	for id, rowCount := range d.queryView.sealedSegmentRowCount {
-		if entry, ok := d.sealedSegments[id]; ok && !entry.Offline {
+		if entry, ok := d.snapshotSegments[id]; ok && !entry.Offline {
```
Comment (czs007) [inline internal/querynodev2/delegator/distribution.go:614]: This is the root cause of the failing `ut-go` (`TestDistribution_UpdateServiceable`, distribution_test.go:914/929/932 — `loadedRatio` expected 0.333, got 0).

This line changed the source of truth for the full `updateServiceable` from `d.sealedSegments` to `d.snapshotSegments`. In production this is equivalent, because the full `updateServiceable` is always called right after `genSnapshot()` (NewDistribution / AddGrowing / SyncTargetVersion / Flush), and `genSnapshot` rebuilds `snapshotSegments` from `sealedSegments`. But `TestDistribution_UpdateServiceable` pokes `d.sealedSegments[...]` directly and then calls `updateServiceable` without a `genSnapshot`, so `snapshotSegments` stays empty and the ratio computes as 0.

Suggested fix (option b, my preference): keep the full `updateServiceable` reading `d.sealedSegments` here — it's equivalent post-`genSnapshot` and matches the "current loaded distribution" intent without breaking the existing test. Only the incremental `updateServiceableByChangedSegments` needs to read `snapshotSegments` (it compares against the just-applied delta). Alternatively (option a) update the test to populate the snapshot shadow state or drive it through the real `AddDistributions` + `genSnapshot` path.
